public/app/features/plugins/extensions/getPluginExtensions.ts

```diff
@@ -22,19 +25,52 @@ import {
   getLinkExtensionPathWithTracking,
 } from './utils';
 
-type GetExtensions = ({
-  context,
-  extensionPointId,
-  limitPerPlugin,
-  addedLinksRegistry,
-  addedComponentsRegistry,
-}: {
-  context?: object | Record<string | symbol, unknown>;
-  extensionPointId: string;
-  limitPerPlugin?: number;
-  addedComponentsRegistry: RegistryType<AddedComponentRegistryItem[]> | undefined;
-  addedLinksRegistry: RegistryType<AddedLinkRegistryItem[]> | undefined;
-}) => { extensions: PluginExtension[] };
+/**
+ * Returns an observable that emits plugin extensions whenever the core extensions registries change.
+ * The observable will emit the initial state of the extensions and then emit again whenever
+ * either the added components registry or the added links registry changes.
+ *
+ * @param options - The options for getting plugin extensions
+ * @returns An Observable that emits the plugin extensions for the given extension point any time the registries change
+ */
+
+export const getObservablePluginExtensions = (
+  options: Omit<GetExtensionsOptions, 'addedComponentsRegistry' | 'addedLinksRegistry'>
+): Observable<ReturnType<GetExtensions>> => {
+  return combineLatest([
+    from(pluginExtensionRegistries.addedComponentsRegistry.getState()),
+    from(pluginExtensionRegistries.addedLinksRegistry.getState()),
+  ]).pipe(
+    switchMap(([initialComponents, initialLinks]) =>
+      combineLatest([
+        merge(of(initialComponents), pluginExtensionRegistries.addedComponentsRegistry.asObservable()),
+        merge(of(initialLinks), pluginExtensionRegistries.addedLinksRegistry.asObservable()),
+      ])
+    ),
+    map(([components, links]) =>
+      getPluginExtensions({
+        ...options,
+        addedComponentsRegistry: components,
+        addedLinksRegistry: links,
+      })
+    ),
+    shareReplay(1)
```

Comment:
I'm a bit uncertain about the `shareReplay` part here without the `refCount`. This means that this observable will never complete/terminate/close. https://rxjs.dev/api/index/function/shareReplay#reference-counting

Reply:
Really good point, removed in https://github.com/grafana/grafana/pull/103063/commits/9e818848d3b9e657df79f67505238d213e98c5d0 👍